note/note_verifier.go

```diff
@@ -17,17 +17,64 @@ package note
 
 import (
 	"crypto/ecdsa"
+	"crypto/ed25519"
 	"crypto/sha256"
 	"crypto/x509"
 	"encoding/base64"
 	"encoding/binary"
+	"errors"
 	"fmt"
 	"strconv"
 	"strings"
 
 	"golang.org/x/mod/sumdb/note"
 )
 
+// SignerVerifierFromEd25519Key returns a note Signer and Verifier given an
+// Ed25519 private key in the standard note-formatted form, e.g.
+// `PRIVATE+KEY+logandmap+38581672+AXJ0FKWOcO2ch6WC8kP705Ed3Gxu7pVtZLhfHAQwp+FE`.
+func SignerVerifierFromEd25519Key(skey string) (note.Signer, note.Verifier, error) {
```

Comment:
How about `NewEd25519SignerVerifier(skey string) (...)`? I think it would fit better with the existing naming style.

Reply:
Yeah, SGTM.